crates/core/component/dex/src/component/dex.rs

```diff
@@ -209,12 +209,26 @@ pub trait StateWriteExt: StateWrite + StateReadExt {
         self.object_put(state_key::config::dex_params_updated(), ())
     }
 
-    fn set_output_data(
+    async fn set_output_data(
         &mut self,
         output_data: BatchSwapOutputData,
         swap_execution_1_for_2: Option<SwapExecution>,
         swap_execution_2_for_1: Option<SwapExecution>,
-    ) {
+    ) -> Result<()> {
+        // Debit the DEX for the swap outflows.
+        // Note that since we credited the DEX for _all_ inflows, we need to debit the
+        // unfilled amounts as well as the filled amounts.
+        self.vcb_debit(Value {
```

Comment:
is the idea that the debit will underflow and cause an error to be returned in the value circuit breaker triggered case? it might be good to explicitly call that out

Reply:
Yep